src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -1,29 +1,90 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.IO;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Extensions;
 using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.Mvc.Internal
 {
     public class FileContentResultExecutor : FileResultExecutorBase
     {
+        // default buffer size as defined in BufferedStream type
+        private const int BufferSize = 0x1000;
+
         public FileContentResultExecutor(ILoggerFactory loggerFactory)
             : base(CreateLogger<FileContentResultExecutor>(loggerFactory))
         {
         }
 
         public Task ExecuteAsync(ActionContext context, FileContentResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            RangeItemHeaderValue range;
+            long rangeLength;
+            bool returnEmptyBody;
+
+            (range, rangeLength, returnEmptyBody) = SetHeadersAndLog(
+                context,
+                result,
+                result.FileContents.Length,
+                result.LastModified,
+                result.EntityTag);
+
+            if (returnEmptyBody)
+            {
+                return Task.CompletedTask;
+            }
+
+            return WriteFileAsync(context, result, range, rangeLength);
         }
 
-        private static Task WriteFileAsync(ActionContext context, FileContentResult result)
+        private static async Task WriteFileAsync(ActionContext context, FileContentResult result, RangeItemHeaderValue range, long rangeLength)
         {
             var response = context.HttpContext.Response;
+            var outputStream = response.Body;
+            var fileContentsStream = new MemoryStream(result.FileContents);
```

Comment:
The reason I did this was to avoid converting long to int to use `Stream.WriteAsync(byte[] buffer, int offset, int count)`